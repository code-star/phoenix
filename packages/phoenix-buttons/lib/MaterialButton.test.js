import * as renderer from 'react-test-renderer';
import {
  containedDefaultMaterialButton,
  containedPrimaryMaterialButton,
  containedSecondaryMaterialButton,
} from './MaterialButton.render';

describe('<MaterialButton />', () => {
  describe('Snaphot', () => {
    test('must match containedDefaultMaterialButton', () => {
      expect(renderer.create(containedDefaultMaterialButton).toJSON()).toMatchSnapshot();
    });

    test('must match containedPrimaryMaterialButton', () => {
      expect(renderer.create(containedPrimaryMaterialButton).toJSON()).toMatchSnapshot();
    });

    test('must match containedSecondaryMaterialButton', () => {
      expect(renderer.create(containedSecondaryMaterialButton).toJSON()).toMatchSnapshot();
    });
  });
});
