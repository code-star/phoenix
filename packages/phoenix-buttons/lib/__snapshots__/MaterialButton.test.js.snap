// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaterialButton /> Snaphot must match containedDefaultMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-raised-15 MuiButton-colorInherit-22"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    Default
  </span>
  <span
    className="MuiTouchRipple-root-30"
  />
</button>
`;

exports[`<MaterialButton /> Snaphot must match containedDisabledMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedSecondary-14 MuiButton-raised-15 MuiButton-raisedSecondary-17"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    Disabled
  </span>
  <span
    className="MuiTouchRipple-root-30"
  />
</button>
`;

exports[`<MaterialButton /> Snaphot must match containedPrimaryMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    Primary
  </span>
  <span
    className="MuiTouchRipple-root-30"
  />
</button>
`;

exports[`<MaterialButton /> Snaphot must match containedSecondaryMaterialButton 1`] = `
<button
  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedSecondary-14 MuiButton-raised-15 MuiButton-raisedSecondary-17"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    Secondary
  </span>
  <span
    className="MuiTouchRipple-root-30"
  />
</button>
`;
